src/stability_sdk/animation.py

```diff
@@ -230,31 +232,54 @@ def __init__(
         # configure Api to retry on classifier obfuscations
         self.api._retry_obfuscation = True
 
-        # 3D accumulate_xforms not ready yet
-        if args.animation_mode == '3D' and args.accumulate_xforms:
-            args.accumulate_xforms = False
-
         self.setup_animation(resume)
 
-    def apply_inpainting(self, mask: np.ndarray, prompts: List[str], weights: List[float], inpaint_steps: int, seed: int):
-        for i in range(len(self.prior_frames)):
-            results = self.api.inpaint(
-                image=self.prior_frames[i],
-                mask=mask,
-                prompts=prompts,
-                weights=weights,
-                steps=inpaint_steps,
-                seed=seed,
-                cfg_scale=self.args.cfg_scale,
-            )
-            self.prior_frames[i] = results[generation.ARTIFACT_IMAGE][0]
+    def build_frame_xform(self, frame_idx) -> matrix.Matrix:
+        args, frame_args = self.args, self.frame_args
+
+        if self.args.animation_mode == '2D':
+            angle = frame_args.angle_series[frame_idx]
+            scale = frame_args.zoom_series[frame_idx]
+            dx = frame_args.translation_x_series[frame_idx]
+            dy = frame_args.translation_y_series[frame_idx]
+            return make_xform_2d(args.width, args.height, math.radians(angle), scale, dx, dy)
+
+        elif self.args.animation_mode == '3D':
+            dx = frame_args.translation_x_series[frame_idx]
+            dy = frame_args.translation_y_series[frame_idx]
+            dz = frame_args.translation_z_series[frame_idx]
+            rx = frame_args.rotation_x_series[frame_idx]
+            ry = frame_args.rotation_y_series[frame_idx]
+            rz = frame_args.rotation_z_series[frame_idx]
+
+            dx, dy, dz = -dx*TRANSLATION_SCALE, dy*TRANSLATION_SCALE, -dz*TRANSLATION_SCALE
+            rx, ry, rz = math.radians(rx), math.radians(ry), math.radians(rz)
+
+            # create xform for the current frame
+            world_view = matrix.multiply(matrix.translation(dx, dy, dz), matrix.rotation_euler(rx, ry, rz))
+            return world_view
+
+        else:
+            return matrix.identity
+
+    def emit_frame(self, frame_idx: int, out_frame: np.ndarray) -> Image.Image:
+        if self.args.save_depth_maps:
+            depth_image = self.generate_depth_image(out_frame)
```

Comment:
Isn't it redundant? We've already calculated the depth map for transform, maybe just reuse it?

Reply:
Yes in the past I spit out the already available depth map but had wound up switching to computing fresh at the end so it matches exactly with the output frame rather than being one warp behind. Definitely can try and optimize this down more as potentially we want depth for pseudo-3D, 3D, VR, and depth output. One issue is wanting this to live in flight in chain and not needing the depth returned to client.